scm/driver/gitlab/webhook.go

```diff
@@ -380,93 +471,76 @@ type (
-			LineCode     string      `json:"line_code"`
-			CommitID     string      `json:"commit_id"`
-			NoteableID   int         `json:"noteable_id"`
-			StDiff       interface{} `json:"st_diff"`
-			System       bool        `json:"system"`
-			UpdatedByID  interface{} `json:"updated_by_id"`
-			Type         string      `json:"type"`
-			Position     struct {
-				BaseSha      string      `json:"base_sha"`
-				StartSha     string      `json:"start_sha"`
-				HeadSha      string      `json:"head_sha"`
-				OldPath      string      `json:"old_path"`
-				NewPath      string      `json:"new_path"`
-				PositionType string      `json:"position_type"`
-				OldLine      interface{} `json:"old_line"`
-				NewLine      int         `json:"new_line"`
-			} `json:"position"`
-			OriginalPosition struct {
-				BaseSha      string      `json:"base_sha"`
-				StartSha     string      `json:"start_sha"`
-				HeadSha      string      `json:"head_sha"`
-				OldPath      string      `json:"old_path"`
-				NewPath      string      `json:"new_path"`
-				PositionType string      `json:"position_type"`
-				OldLine      interface{} `json:"old_line"`
-				NewLine      int         `json:"new_line"`
-			} `json:"original_position"`
-			ResolvedAt     interface{} `json:"resolved_at"`
-			ResolvedByID   interface{} `json:"resolved_by_id"`
-			DiscussionID   string      `json:"discussion_id"`
-			ChangePosition struct {
-				BaseSha      interface{} `json:"base_sha"`
-				StartSha     interface{} `json:"start_sha"`
-				HeadSha      interface{} `json:"head_sha"`
-				OldPath      interface{} `json:"old_path"`
-				NewPath      interface{} `json:"new_path"`
-				PositionType string      `json:"position_type"`
-				OldLine      interface{} `json:"old_line"`
-				NewLine      interface{} `json:"new_line"`
-			} `json:"change_position"`
-			ResolvedByPush interface{} `json:"resolved_by_push"`
-			URL            string      `json:"url"`
+		ObjectAttributes      struct {
+			ID               int         `json:"id"`
+			Note             string      `json:"note"`
+			NoteableType     string      `json:"noteable_type"`
+			AuthorID         int         `json:"author_id"`
+			CreatedAt        string      `json:"created_at"`
+			UpdatedAt        string      `json:"updated_at"`
+			ProjectID        int         `json:"project_id"`
+			Attachment       interface{} `json:"attachment"`
+			LineCode         string      `json:"line_code"`
+			CommitID         string      `json:"commit_id"`
+			NoteableID       int         `json:"noteable_id"`
+			StDiff           interface{} `json:"st_diff"`
+			System           bool        `json:"system"`
+			ResolvedAt       interface{} `json:"resolved_at"`
+			ResolvedByID     interface{} `json:"resolved_by_id"`
+			ResolvedByPush   interface{} `json:"resolved_by_push"`
+			DiscussionID     string      `json:"discussion_id"`
+			URL              string      `json:"url"`
+			Position         interface{} `json:"position"`
```

Comment:
Why are we changing Position to an interface? Same for others as well

Reply:
There is a mismatch between the datatypes used for Position in this file previously and the gitlab go package https://github.com/go-playground/webhooks/blob/v5.17.0/gitlab/payload.go#L322. 

1. It doesn't contain extra fields which are present in Position https://github.com/go-playground/webhooks/blob/59fb74817c3d5c702553964f4d2faff80f23d350/gitlab/payload.go#L385 which can lead to unmarshal failing
2. Position/OriginalPosition/ChangedPosition have the same type in the go package but different here

Ideally we should keep the Position type same but kept interface here to be on the safer side. I could not find documentation on what the possible values could be each field.